lib/problems.rb

```diff
@@ -1,13 +1,69 @@
 require_relative './stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n), push and pop should be O(1) operations since the linked list has both a head and tail pointer
+# Space Complexity: O(n) if the string is not balanced and comprised wholly of open brackets, also O(n) if it actually closes
 def balanced(string)
```

Comment:
This works, but you can make it simpler by using a hash to do the mashing instead of using the case statement.